src/main/java/com/slack/kaldb/logstore/LogMessage.java

```diff
@@ -142,4 +142,17 @@ public void addProperty(String key, Object value) {
   public LogWireMessage toWireMessage() {
     return new LogWireMessage(getIndex(), getType(), id, source);
   }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+    LogMessage that = (LogMessage) o;
+    return timeSinceEpochMilli == that.timeSinceEpochMilli;
```

Comment:
This doesn't seem right - we would expect to see multiple messages with identical timestamps.

Reply:
it does not! Unsure what else fields can I use to compare?

Reply:
Per our discussion, `_source` will be expected to contain a significant amount of unique attributes. We probably should ensure this map is the same across objects.